python/dgl/runtime/scheduler.py

```diff
@@ -74,25 +75,37 @@ def schedule_recv(graph,
     inplace: bool
         If True, the update will be done in place
     """
-    src, dst, mid = graph._msg_graph.in_edges(recv_nodes)
-    if len(mid) == 0:
-        # All recv nodes are 0-degree nodes; downgrade to apply nodes.
+    _, _, eid = graph._graph.in_edges(recv_nodes)
+    if len(eid) > 0:
+        e_mask = graph._msg_index.get_items(eid).nonzero()
+        eid = eid.get_items(e_mask)
```

Comment:
I guess this does not work for MX. MX does not support using a bool mask to index the value.

Reply:
Oh, it's not a boolean mask. Instead just nonzero index. Let me rename `e_mask` to `nonzero_idx`